src/API.jl

```diff
@@ -946,6 +946,20 @@ function precompile(ctx::Context; internal_call::Bool=false)
         was_recompiled[pkgid] = false
     end
     
+    # guarding against circular deps
+    function in_deps(pkg, deps, dmap)
+        length(deps) == 0 && return false
+        pkg in deps && return true
+        return any(map(dep->in_deps(pkg, dmap[dep], dmap), deps))
+    end
+    for (pkg, deps) in depsmap
+        if in_deps(pkg, deps, depsmap)
+            notify(was_processed[pkg])
+            Operations.precomp_suspend!(pkg)
+            !internal_call && @warn "Circular dependency detected. Precompilation skipped for $pkg"
```

Comment:
Should pkg be interpolated or should it be a log message argument?
In general I am never clear on that.
It seems unlikely someone will be using va structured log consumer with Pkg, but. Idk, may a Pkg graphical front-enf might be made

Reply:
I don't think I'm the person to answer this. Should it hold up merge?